### ISSUE TITLE
Fix procedure doesn't recognize memory limits

### ISSUE THREAD

**Issue by chrischabot** [memgraph/memgraph#1219]
**Memgraph version**
2.10.2

**Environment**
AWS EC2 Ubuntu 22 running memgraph-platform in a docker container

**Describe the bug**
I run the following query on BlueJ to do community detection:

```
CALL community_detection.get()
YIELD node, community_id
WITH node, community_id
WHERE 'Person' IN labels(node)
SET node.community_id = community_id;

```
And the DB size on a fresh start uses 13 out of 28 gigs of memory

Once I run the query, it'll execute and allocate memory until all of the 28 Gb avalable is taken, and keeps going -- up until there's no memory left for allocating file / network handles, so ssh and other services become unavailable. 

The only solution is to reboot the server as it'll stay in this state for multiple days

**To Reproduce**

Download the latest bluej db snapshop from https://bluej.memgraph.com/dump/bluej-latest.cypherl.gz and run the query on a linux host (& other platforms presumably) with 32 gb of ram. The machine will crash

**Expected behavior**
No crash. Crash bad

Screenshots of start of query & moment the server became unavailable:

![CleanShot 2023-09-02 at 19 54 16@2x](https://github.com/memgraph/memgraph/assets/11455514/f64f3d79-b5b1-44da-baa9-e2ff588e4570)
![CleanShot 2023-09-02 at 19 46 19@2x](https://github.com/memgraph/memgraph/assets/11455514/df2dcb64-1c44-49ff-94b8-ae45a1e4a082)

**Comment by gitbuda** [comment]
Thanks for reporting! Added to the issues project, we'll take a look ASAP, probably on Monday we'll have some info 😃

**Comment by antoniofilipovic** [comment]
The problem with this issue is twofold, one part is part of the issue on MAGE, other is on Memgraph. 


For memgraph issue is that a procedure doesn't recognize memory limits.

For MAGE issue is that community detection does not release memory while it is processing, although I am not sure what it means. I will split this issue into two parts, open issue on MAGE for community detection and keep here what is part of memgraph.

**Comment by as51340** [comment]
Closing since it is fixed with 2.12
